pkg/kvstore/backend.go

```diff
@@ -153,4 +153,12 @@ type BackendOperations interface {
 
 	// Decodes a key previously encoded back into the original binary slice
 	Decode(in string) ([]byte, error)
+
+	// ListAndWatch creates a new watcher which will watch the specified
+	// prefix for changes. Before doing this, it will list the current keys
+	// matching the prefix and report them as new keys. Name can be set to
+	// anything and is used for logging messages. The Events channel is
+	// created with the specified sizes. Upon every change observed, a
+	// KeyValueEvent will be sent to the Events channel
+	ListAndWatch(name, prefix string, chanSize int) *Watcher
```

Comment:
Is the initial list synchronous? (it isn't) It might be worth mentioning that since it returns an object. Is there a delay before the `Watcher` is ready or reports events?

Reply:
Also, do we test this behaviour somewhere? I couldn't find them in the tests.

Reply:
The initial list is not sync for ipcache. I haven't changed the watch logic of ipcache. I'm only making it callable via multiple clients.

That said, the test is here:

```
func (s *BaseTests) TestListAndWatch(c *C) {
        key1, key2 := "foo2/key1", "foo2/key2"
        val1, val2 := []byte("val1"), []byte("val2")

        DeletePrefix("foo2/")
        defer DeletePrefix("foo2/")

        err := CreateOnly(key1, val1, false)
        c.Assert(err, IsNil)

        w := ListAndWatch("testWatcher2", "foo2/", 100)
        c.Assert(c, Not(IsNil))

        expectEvent(c, w, EventTypeCreate, key1, val1)
        expectEvent(c, w, EventTypeListDone, "", []byte{})
```